### ISSUE TITLE
3.9 will break registry-route addon

### ISSUE THREAD

**Issue by jorgemoralespou** [minishift/minishift#2140]
```
$ minishift addon apply registry-route
-- Applying addon 'registry-route':.Error applying the add-on: Error executing command 'ssh command error:
command : docker exec -t origin /usr/bin/openshift admin ca create-server-cert --signer-cert=/var/lib/origin/openshift.local.config/master/ca.crt --signer-key=/var/lib/origin/openshift.local.config/master/ca.key --signer-serial=/var/lib/origin/openshift.local.config/master/ca.serial.txt --hostnames='docker-registry-default.192.168.64.44.nip.io,docker-registry-default.192.168.64.44.nip.io:443,docker-registry.default.svc.cluster.local,172.30.1.1' --cert=/etc/secrets/registry.crt --key=/etc/secrets/registry.key
err     : exit status 1
output  : Error: unknown command "admin" for "openshift"
Run 'openshift --help' for usage.

$ minishift version
minishift v1.14.0+1ec5877

$ minishift openshift version
openshift v3.9.0-alpha.3+78ddc10
kubernetes v1.9.1+a0ce1bc657
etcd 3.2.8
```

**Comment by LalatenduMohanty** [comment]
@jorgemoralespou Thanks, do you know when it is coming out?

**Comment by jorgemoralespou** [comment]
End of month

**Comment by anjannath** [comment]
I did some poking around and it seems the `openshift admin` command is no there anymore, we have to use `oc adm`, `oc cli`.

We need to run these commands from inside the origin container, right now `oc` is not run inside the origin container AFAIK.

**Comment by budhrg** [comment]
> I did some poking around and it seems the openshift admin command is not there anymore, we have to use oc adm, oc create.

If this is the case, we also need to think about the backward compatibility of the addon as how we are going to support it for version < 3.9.0.

**Comment by anjannath** [comment]
> we also need to think about the backward compatibility of the addon as how we are going to support it for version < 3.9.0

Yes, and also the parts that interact with OpenShift. #2156 is also caused because of this, probably we need to check all the sub-commands of `minishift openshift`.

**Comment by praveenkumar** [comment]
Looks they removed some of the commands from the openshift binary. We need to find an alternative for this now.

```
docker@minishift:~$ docker run --rm -it openshift/origin:v3.9.0-alpha.3 help
OpenShift Application Platform 

The OpenShift distribution of Kubernetes helps you build, deploy, and manage your applications on top of Docker
containers. To start an all-in-one server with the default configuration, run: 

  $ openshift start &

Available Commands:
  completion  Output shell completion code for the specified shell (bash or zsh)
  help        Help about any command
  start       Launch all-in-one server
  version     Display client and server versions

Use "openshift <command> --help" for more information about a given command.
Use "openshift options" for a list of global command-line options (applies to all commands).

docker@minishift:~$ docker run --rm -it openshift/origin:v3.7.1 help
OpenShift Application Platform 

The OpenShift distribution of Kubernetes helps you build, deploy, and manage your applications on top of Docker
containers. To start an all-in-one server with the default configuration, run: 

  $ openshift start &

Available Commands:
  admin       Tools for managing a cluster
  cli         Command line tools for managing applications
  completion  Output shell completion code for the specified shell (bash or zsh)
  ex          Experimental commands under active development
  help        Help about any command
  kube        Kubernetes cluster management via kubectl
  start       Launch all-in-one server
  version     Display client and server versions

Use "openshift <command> --help" for more information about a given command.
Use "openshift options" for a list of global command-line options (applies to all commands).
```

**Comment by anjannath** [comment]
We were using the `openshift admin` and `openshift ex` commands.. both are removed and now present in `oc`.

So, maybe we need to add code to run `oc` commands from inside the origin container.

**Comment by budhrg** [comment]
> Looks they removed some of the commands from the openshift binary. We need to find an alternative for this now.

Too much change.

**Comment by LalatenduMohanty** [comment]
> @LalatenduMohanty any idea? is this due to the rewrite?

I do not think so. But I also think OpenShift developers do not know we use the OpenSHift server binary. The general understanding is that we use `oc cluster up`.

**Comment by praveenkumar** [comment]
@gbraad This is mostly the rewrite since they want to trim `openshift` server binary and remove all the extra commands which can be done by client binary.

**Comment by gbraad** [comment]
So, in future all if these commands are in `oc` guaranteed?

**Comment by LalatenduMohanty** [comment]
Resolved by #2161
